crown-court-contribution/src/main/java/uk/gov/justice/laa/crime/contribution/service/MaatCalculateContributionService.java

```diff
@@ -262,8 +262,8 @@ public ApiMaatCalculateContributionResponse doContribs(final CalculateContributi
     }
 
     private boolean shouldCreateContribs(ContributionResult result, CalculateContributionDTO calculateContributionDTO) {
-        return (result.monthlyAmount() != null && result.monthlyAmount()
-                .compareTo(calculateContributionDTO.getMonthlyContributions()) != 0)
+        return (result.monthlyAmount() != null && null != calculateContributionDTO.getMonthlyContributions()
+                && result.monthlyAmount().compareTo(calculateContributionDTO.getMonthlyContributions()) != 0)
```

Comment:
Can we move these changes to the builder class please? Otherwise we may need to add additional null checks in the future. Better to default the values to 0 when we receive the initial request.